### PR TITLE
fix Join.vue 'gi.contracts/group/proposal' error + more payments plumming

### PR DESCRIPTION
**Pull request by taoeffect**

### CODE REVIEW

File: frontend/model/contracts/group.js
Comment (sandrina-p) [inline frontend/model/contracts/group.js:166]: Why this validation is done during process instead of validate as before?

File: frontend/model/state.js
Comment (sandrina-p) [inline frontend/model/state.js:251]: why can't we pass groupId here?

File: frontend/model/contracts/group.js
Comment (taoeffect) [reply frontend/model/contracts/group.js:166]: Because we cannot access state that exists outside of the contract (submodule) state, and `sbp('state/vuex/getters').ourUsername` does that

**Review by sandrina-p — APPROVED** [review]

File: frontend/model/state.js
Comment (taoeffect) [reply frontend/model/state.js:251]: because right now getters cannot be access from the contract safely, and i'm working on making it possible to do that... i'll explain in more detail offline

**Comment by taoeffect** [comment]
**NOTE: this is an exceptional merge that violates Rule 4 of our contributions guidelines because it's during a hackathon and I have approval from @sandrina-p to merge it, and @sandrina-p cannot merge it herself because the build is failing, but it's an error unrelated to this PR.**

**Pull request merged**
